// @progress/kendo-react-layout/dist/npm/panelbar/services/navigationService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @hidden
 */
var NavigationAction;
(function (NavigationAction) {
    NavigationAction[NavigationAction["Undefined"] = 0] = "Undefined";
    NavigationAction[NavigationAction["Toggle"] = 1] = "Toggle";
    NavigationAction[NavigationAction["Next"] = 2] = "Next";
    NavigationAction[NavigationAction["Previous"] = 3] = "Previous";
})(NavigationAction = exports.NavigationAction || (exports.NavigationAction = {}));
/**
 * @hidden
 */
var NavigationService = /** @class */ (function () {
    function NavigationService(onNavigate) {
        this.onNavigate = onNavigate;
    }
    NavigationService.prototype.navigate = function (event, rtl) {
        var keyCode = event.keyCode;
        var action = NavigationAction.Undefined;
        switch (keyCode) {
            case 37 /* left */:
                action = !rtl ? NavigationAction.Previous : NavigationAction.Next;
                break;
            case 38 /* up */:
                action = NavigationAction.Previous;
                break;
            case 39 /* right */:
                action = !rtl ? NavigationAction.Next : NavigationAction.Previous;
                break;
            case 40 /* down */:
                action = NavigationAction.Next;
                break;
            case 32 /* space */:
                action = NavigationAction.Toggle;
                break;
            case 13 /* enter */:
                action = NavigationAction.Toggle;
                break;
            default:
                action = NavigationAction.Undefined;
                break;
        }
        if (action !== NavigationAction.Undefined) {
            event.preventDefault();
        }
        this.onNavigate(action);
    };
    return NavigationService;
}());
exports.default = NavigationService;
