// @progress/kendo-react-layout/dist/npm/tabstrip/TabStripContent.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var PropTypes = require("prop-types");
var kendo_react_animation_1 = require("@progress/kendo-react-animation");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var TabStripContent = /** @class */ (function (_super) {
    __extends(TabStripContent, _super);
    function TabStripContent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * @hidden
     */
    TabStripContent.prototype.render = function () {
        var contentClasses = kendo_react_common_1.classNames('k-content', 'k-state-active');
        return (React.createElement("div", { className: contentClasses, style: this.props.style }, this.renderContent(this.props.children)));
    };
    TabStripContent.prototype.renderContent = function (children) {
        var contentProps = {
            'role': 'tabpanel',
            'aria-expanded': true
        };
        var animationStyle = {
            position: 'initial'
        };
        var tab = React.Children.toArray(children)[this.props.selected];
        var child = this.props.animation
            ? (React.createElement(kendo_react_animation_1.Fade, { style: animationStyle, appear: true },
                React.createElement("div", __assign({}, contentProps, { key: this.props.index }), tab.props.children)))
            : (React.createElement("div", __assign({}, contentProps, { key: this.props.index }), tab.props.children));
        return child;
    };
    /**
     * @hidden
     */
    TabStripContent.propTypes = {
        animation: PropTypes.bool,
        children: PropTypes.oneOfType([
            PropTypes.element,
            PropTypes.arrayOf(PropTypes.element)
        ]),
        selected: PropTypes.number,
        style: PropTypes.object
    };
    return TabStripContent;
}(React.Component));
exports.default = TabStripContent;
